plugins/inputs/azure_monitor/azure_monitor_test.go

```diff
@@ -0,0 +1,1002 @@
+package azure_monitor
+
+import (
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/monitor/armmonitor"
+	"github.com/influxdata/telegraf/testutil"
+	receiver "github.com/logzio/azure-monitor-metrics-receiver"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInit_ResourceTargetsOnly(t *testing.T) {
+	am := &AzureMonitor{
+		SubscriptionID: testSubscriptionID,
+		ClientID:       testClientID,
+		ClientSecret:   testClientSecret,
+		TenantID:       testTenantID,
+		ResourceTargets: []*ResourceTarget{
+			{
+				testResourceGroup1ResourceType1Resource1,
+				[]string{testMetric1, testMetric2, testMetric3},
+				[]string{string(armmonitor.AggregationTypeEnumTotal), string(armmonitor.AggregationTypeEnumAverage)},
+			},
+			{
+				testResourceGroup1ResourceType2Resource2,
+				[]string{testMetric1, testMetric2},
+				[]string{},
+			},
+			{
+				testResourceGroup2ResourceType1Resource3,
+				[]string{},
+				[]string{string(armmonitor.AggregationTypeEnumTotal)},
+			},
+			{
+				testResourceGroup1ResourceType2Resource2,
+				[]string{},
+				[]string{},
+			},
+			{
+				testResourceGroup1ResourceType1Resource1,
+				[]string{},
+				[]string{},
+			},
+		},
+		Log:          testutil.Logger{},
+		azureClients: setMockAzureClients(),
+	}
+
+	for index := 1; index <= 27; index++ {
+		if index <= 10 {
+			am.ResourceTargets[4].Metrics = append(am.ResourceTargets[4].Metrics, testMetric1)
+		} else if index <= 23 {
+			am.ResourceTargets[4].Metrics = append(am.ResourceTargets[4].Metrics, testMetric2)
+		} else {
+			am.ResourceTargets[4].Metrics = append(am.ResourceTargets[4].Metrics, testMetric3)
+		}
+	}
+
+	expectedResourceMetrics := make([]string, 0)
+	for index := 1; index <= receiver.MaxMetricsPerRequest; index++ {
+		if index <= 10 {
+			expectedResourceMetrics = append(expectedResourceMetrics, testMetric1)
+		} else {
+			expectedResourceMetrics = append(expectedResourceMetrics, testMetric2)
+		}
+	}
+
+	err := am.Init()
+	require.NoError(t, err)
+
+	assert.Len(t, am.receiver.Targets.ResourceTargets, 8)
```

Comment:
Please prefer `require` over `assert`.
```suggestion
	require.Len(t, am.receiver.Targets.ResourceTargets, 8)
```